packages/web/src/common/store/social/tracks/sagas.ts

```diff
@@ -53,22 +53,26 @@ export function* watchRepostTrack() {
 export function* repostTrackAsync(
   action: ReturnType<typeof socialActions.repostTrack>
 ) {
-  yield* waitForWrite()
+  yield* call(waitForWrite)
   const userId = yield* select(getUserId)
   if (!userId) {
     yield* put(signOnActions.openSignOn(false))
     yield* put(signOnActions.showRequiresAccountModal())
     yield* put(make(Name.CREATE_ACCOUNT_OPEN, { source: 'social action' }))
     return
   }
-  if (userId === action.trackId) {
-    return
-  }
 
   // Increment the repost count on the user
   const user = yield* select(getUser, { id: userId })
   if (!user) return
 
+  const track = yield* select(getTrack, { id: action.trackId })
+  if (!track) return
+
+  if (track.owner_id === userId) {
```

Comment:
cc: @raymondjacobson re: last week's msg

Reply:
yeah this seems correct!